lib/smci_report.rb

```diff
@@ -0,0 +1,205 @@
+# Exports all publications for the given sunets or cap_profile_ids in an 'approved' state
+#  and it also harvests publications by name for authors who do not have a sunet or cap_profile_id
+# This is a custom report for Stanford Medicine Center for Improvement (SMCI)
+#   specified in https://github.com/sul-dlss/sul_pub/issues/1201
+
+# October 2020
+
+# You must have an input csv file with the following columns (including a header row):
+#   sunetid,cap_profile_id,first_name,middle_name,last_name,institutions
+#
+#  For authors that exist in profiles, you can specify either a sunet or cap_profile_id (sunet preferred if both specified)
+#  For authors that do NOT exist in profiles, you MUST leave the cap_profile_id and sunet field columns blank, and then you must specify first name and last name
+#    and optionally middle name a comma delimtied list of institutions (defaults to 'stanford' if no institutions provided)
+
+# Run with:
+#
+# smci = SMCIReport.new(input_file: 'tmp/input.csv', output_file: 'tmp/output_file.csv', date_since: '1/1/2000', time_span: '1year')
```

Comment:
What is the context for executing these commands?

Reply:
Also, mention that this is available as a rake task.

Reply:
right, this was before there was a rake task...but I will indicate you can run this from the Rails console as well if you want